qa/L0_backend_python/response_sender/response_sender_test.py

```diff
@@ -0,0 +1,293 @@
+            grpcclient.InferInput(
+                "SEND_COMPLETE_FINAL_FLAG_AFTER_RETURN", shape, "BOOL"
+            ),
+        ]
+        inputs[0].set_data_from_numpy(
+            np.array([[number_of_response_before_return]], np.uint8)
+        )
+        inputs[1].set_data_from_numpy(
+            np.array([[send_complete_final_flag_before_return]], bool)
+        )
+        inputs[2].set_data_from_numpy(np.array([[return_a_response]], bool))
+        inputs[3].set_data_from_numpy(
+            np.array([[number_of_response_after_return]], np.uint8)
+        )
+        inputs[4].set_data_from_numpy(
+            np.array([[send_complete_final_flag_after_return]], bool)
+        )
+        return inputs
+
+    def _generate_streaming_callback_and_responses_pair(self):
+        responses = []  # [{"result": result, "error": error}, ...]
+
+        def callback(result, error):
+            responses.append({"result": result, "error": error})
+
+        return callback, responses
+
+    def _infer(
+        self,
+        model_name,
+        number_of_response_before_return,
+        send_complete_final_flag_before_return,
+        return_a_response,
+        number_of_response_after_return,
+        send_complete_final_flag_after_return,
+    ):
+        inputs = self._get_inputs(
+            number_of_response_before_return,
+            send_complete_final_flag_before_return,
+            return_a_response,
+            number_of_response_after_return,
+            send_complete_final_flag_after_return,
+        )
+        callback, responses = self._generate_streaming_callback_and_responses_pair()
+        with grpcclient.InferenceServerClient("localhost:8001") as client:
+            client.start_stream(callback)
+            client.async_stream_infer(model_name, inputs)
+            time.sleep(2)  # to collect all responses
+            client.stop_stream()
+        return responses
+
+    def _assert_responses_valid(
+        self,
+        responses,
+        number_of_response_before_return,
+        return_a_response,
+        number_of_response_after_return,
+    ):
+        before_return_response_count = 0
+        response_returned = False
+        after_return_response_count = 0
+        for response in responses:
+            result, error = response["result"], response["error"]
+            self.assertIsNone(error)
+            result_np = result.as_numpy(name="INDEX")
+            response_id = result_np.sum() / result_np.shape[0]
+            if response_id < 1000:
+                self.assertFalse(
+                    response_returned,
+                    "Expect at most one response returned per request.",
+                )
+                response_returned = True
+            elif response_id < 2000:
+                before_return_response_count += 1
+            else:  # response_id < 3000
+                after_return_response_count += 1
```

Comment:
Can you explicitly fail for an unexpected response_id for future test change sanity checks?

```python
if response_id < 1000:
    # ...
elif response_id < 2000:
    # ...
elif response_id < 3000:
    # ...
else: 
    raise ValueError(f"Unexpected response_id: {response_id}")
```

Reply:
Good idea! Updated.